compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/input/pointer/HitPathTracker.kt

```diff
@@ -388,22 +388,21 @@ internal class Node(val pointerInputFilter: PointerInputFilter) : NodeParent() {
                 @Suppress("DEPRECATION")
                 isIn = !enterExitChange.isOutOfBounds(size)
             }
-            if (isIn != wasIn &&
-                (
-                    event.type == PointerEventType.Move ||
-                        event.type == PointerEventType.Enter ||
-                        event.type == PointerEventType.Exit
-                    )
+            if (event.type == PointerEventType.Move ||
+                event.type == PointerEventType.Enter ||
+                event.type == PointerEventType.Exit
             ) {
-                event.type = if (isIn) {
-                    PointerEventType.Enter
-                } else {
-                    PointerEventType.Exit
+                event.type = when {
```

Comment:
Rather hard to read logic.
Maybe smth like
```
if (event.type == PointerEventType.Move) {
    event.type = when {
       !wasIn && isIn && !hasEntered -> PointerEventType.Enter
       wasIn && !isIn && hasEntered -> PointerEventType.Exit
       else -> PointerEventType.Move
}
```

Reply:
Thanks, looks better.

But I can't remove `event.type == PointerEventType.Exit`. The logic is asymmetric, unfortunately.

Reply:
Why? Cannot we force enter as well via keyboard navigation?

Reply:
> force enter as well via keyboard navigation?

Enter is pure pointer/mouse thing.

But I realized there is no special case, I just mask a bug from the desktop code, where we don't properly check the owner's bounds.

Fixed it in https://github.com/JetBrains/androidx/pull/109, and refactored this PR.